src/group_theory/free_group.lean

```diff
@@ -467,6 +496,18 @@ set.subset.antisymm
 
 end to_group
 
+section
+variables (X : Type*) (G : Type*) [group G]
+
+/-- The bijection underlying the free-forgetful adjunction for abelian groups. -/
+@[simps]
+def hom_equiv : (free_group X →* G) ≃ (X → G) :=
```

Comment:
For things like the free algebra, we call this `lift` and write it with the direction swapped.

Reply:
The convention I took came from the free/forgetful adjunction for abelian groups (which I also took as a boilerplate for the abelianization adjunction). I agree that it would be good to unify the names of lemmas throughout the different free/forgetful adjunctions. However, I don't think, this should be part of this PR and there should first be some agreement what good conventions are. Note that `lift` in `free_abelian_group` is what is called `lift_aux` in `free_algebra` if I understand correctly.

Reply:
> Note that lift in free_abelian_group is what is called lift_aux in free_algebra if I understand correctly.

Yes, I haven't gotten around to converting all of the `lift` functions to equivalences yet ;)

Reply:
> However, I don't think, this should be part of this PR and there should first be some agreement what good conventions are

I think `lift` is fairly established terminology - it certainly seems to appear more frequently than `hom_equiv`, so it would be nice to use that terminology in this PR.

Reply:
Since I don't really care, I changed it to lift, but note that there are still quite a few instances in the library where this is called `hom_equiv` and that also makes sense at it is exactly the name of the field we are trying to fill in for the adjunction.